src/compiler/factory.ts

```diff
@@ -1618,17 +1618,23 @@ namespace ts {
         );
     }
 
-    function createReactNamespace(reactNamespace: string, parent: JsxOpeningLikeElement) {
+    export function createJsxFactory(jsxFactory: string) {
+        // No explicit validation of this parameter is required. Users are

```

Comment:
I need some help with deciding what strategy is best for this.

To validate a dotted name as a `QualifiedName`, there’s a few ways I can see to do it:

- basic regex or string manipulation (e.g. `.split(‘.’)`)
- creating a mini parser using scanner, iterating over the tokens and asserting it’s a sequence of `Identifier (Dot, Identifier)*`
- using the parser (`createSourceFile`) to parse the string `a.b.c` and using the resulting source file to both assert the structure, and to extract the `a` identifier to mark it as used
- …?

Balancing simplicity with correctness is the challenge here, I'm not sure what the preferred bias is.